tsdb/engine/wal/wal.go

```diff
@@ -0,0 +1,1591 @@
+/*
+Package WAL implements a write ahead log optimized for write throughput
+that can be put in front of the database index.
+
+The WAL is broken into different partitions. The default number of
+partitions is 5. Each partition consists of a number of segment files.
+By default these files will get up to 2MB in size before a new segment
+file is opened. The files are numbered and start at 1. The number
```

Comment:
Not a big deal, but it was surprising to me. For example Apache Kafka numbers its partitions (and files on disk) start with 0.